lib/api.ts

```diff
@@ -53,132 +22,159 @@ export interface MemberPublicEditing extends MemberPublic {
+export interface DocumentData {
+  id: string;
+  fields: any;
+}
 
-        const focusLookup = () => {
-          const memberFocus = member.fields["Focus"];
-          if (memberFocus && Array.isArray(memberFocus)) {
-            return memberFocus.map((foc) => {
-              return (
-                focuses
-                  .filter((thisFoc) => foc === thisFoc.id)
-                  .map((foc) => {
-                    return {
-                      name:
-                        typeof foc.fields["Name"] === "string"
-                          ? foc.fields["Name"]
-                          : null,
-                      id:
-                        typeof foc.fields["ID"] === "string"
-                          ? foc.fields["ID"]
-                          : null,
-                    };
-                  })[0] || null
-              );
-            });
-          }
-          return null;
-        };
+export async function getFirebaseTable(table: string) {
+  const documentsCollection = collection(db, table);
+  const documentsSnapshot = await getDocs(documentsCollection);
+  const documentsData = documentsSnapshot.docs.map((doc) => ({
+    id: doc.id,
+    fields: doc.data(),
+  }));
+  return documentsData;
+}
 
-        const industryLookup = () => {
-          const memberIndustry = member.fields["Industry"];
-          if (memberIndustry && Array.isArray(memberIndustry)) {
-            return memberIndustry.map((ind) => {
-              return (
-                industries
-                  .filter((thisInd) => ind === thisInd.id)
-                  .map((ind) => {
-                    return {
-                      name:
-                        typeof ind.fields["Name"] === "string"
-                          ? ind.fields["Name"]
-                          : null,
-                      id:
-                        typeof ind.fields["ID"] === "string"
-                          ? ind.fields["ID"]
-                          : null,
-                    };
-                  })[0] || null
-              );
-            });
-          }
-          return null;
-        };
+function regionLookup(member: DocumentData, regions: DocumentData[]) {
+  return (
+    regions.find((region) => {
+      const memberRegion = member.fields.regions;
+      // TODO: Change this check if we're expecting only one region per member
+      if (
+        memberRegion &&
+        Array.isArray(memberRegion) &&
+        memberRegion.length !== 0
+      ) {
+        return region.id === memberRegion[0].id;
+      }
+    })?.fields.name || null
+  );
+}
 
-        const emailLookup = () => {
-          const memberEmail = member.fields["Email"];
-          if (member.fields["Email"] === undefined) {
-            return null;
-          }
-          const [first, last, domain] = useEmailCloaker(memberEmail);
-          return [first, last, domain];
-        };
+function focusLookup(member: DocumentData, focuses: DocumentData[]) {
```

Comment:
Probably a future update, but adding return types to functions would really help with maintenance so the compiler tells you something is wrong rather than finding out at runtime.

Reply:
ah yup never added return type here, great catch, will add it in the future